examples/java-matter-controller/java/src/com/matter/controller/Main.java

```diff
@@ -18,82 +18,91 @@
+import com.matter.controller.commands.common.*;
+import com.matter.controller.commands.discover.*;
+import com.matter.controller.commands.pairing.*;
+import java.util.ArrayList;
 
 public class Main {
-  private static void ShowUsage(Command[] commands) {
-    StringBuffer arguments = new StringBuffer();
-    StringBuffer attributes = new StringBuffer();
-
-    for (Command command : commands) {
-      arguments.append("    ");
-      arguments.append(command.getName());
-
-      int argumentsCount = command.getArgumentsCount();
-      for (int j = 0; j < argumentsCount; j++) {
-        arguments.append(" ");
-        arguments.append(command.getArgumentName(j));
-      }
-
-      arguments.append("\n");
-
-      if ("read".equals(command.getName()) && command.getAttribute().isPresent()) {
-        attributes.append("    " + command.getAttribute().get() + "\n");
-      }
-    }
-
-    System.out.println(
-        String.format(
-            "Usage: \n"
-                + "  java_matter_controller command [params]\n\n"
-                + "  Supported commands and their parameters:\n%s\n"
-                + "  Supported attribute names for the 'read' command:\n%s",
-            arguments, attributes));
+  private static void registerCommandsDiscover(
+      CommandManager commandManager, CredentialsIssuer credentialsIssuer) {
+    ArrayList<Command> clusterCommands = new ArrayList<Command>();
+    DiscoverCommand discoverCommand = new DiscoverCommand(credentialsIssuer);
+    DiscoverCommissionablesCommand discoverCommissionablesCommand =
+        new DiscoverCommissionablesCommand(credentialsIssuer);
+    DiscoverCommissionersCommand discoverCommissionersCommand =
+        new DiscoverCommissionersCommand(credentialsIssuer);
+    clusterCommands.add(discoverCommand);
+    clusterCommands.add(discoverCommissionablesCommand);
+    clusterCommands.add(discoverCommissionersCommand);
+
+    commandManager.register("discover", clusterCommands);
   }
 
-  private static void runCommand(CredentialsIssuer credIssuerCmds, String[] args) {
-
-    // TODO::Start list of available commands, this hard coded list need to be replaced by command
-    // registration mechanism.
-    Command[] commands = {new On(credIssuerCmds), new Off(credIssuerCmds)};
-    // End list of available commands
-
-    if (args.length == 0) {
-      ShowUsage(commands);
-      return;
-    }
-
-    for (Command cmd : commands) {
-      if (cmd.getName().equals(args[0])) {
-        String[] temp = Arrays.copyOfRange(args, 1, args.length);
-
-        try {
-          cmd.initArguments(args.length - 1, temp);
-          cmd.run();
-        } catch (IllegalArgumentException e) {
-          System.out.println("Arguments init failed with exception: " + e.getMessage());
-        } catch (Exception e) {
-          System.out.println("Run command failed with exception: " + e.getMessage());
-        }
-        break;
-      }
-    }
+  private static void registerCommandsPairing(
```

Comment:
Would it make sense to use set multibindings here? Could make this a lot easier: https://dagger.dev/dev-guide/multibindings.html

Reply:
looking at this feature, but it seems we first need to integrate this package since it is not available by default.

Reply:
After some pilots, I believe Dagger 2 injection could help us simplify some objection construction,but this lib is not available in IntelliJ IDEA free version, I need to purchase the license to use this feature.

```
Not compatible with the version of your running IDE (IntelliJ IDEA 2022.2.3)
Compatible with IntelliJ IDEA (Ultimate, Community, Educational),
```
And we also need to figure out how to integrate this library together with JSR 330 to GN build system to use its features. I would prefer not spend effort on this at this moment since we will transit to Kotlin soon.

Reply:
Sounds good! How weird that you're getting intellij license issues